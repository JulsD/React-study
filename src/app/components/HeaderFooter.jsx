import React from 'react';
import SearchNav from './SearchNav';
import MovieInfo from './MovieInfo';

import common from './../../styles/common.css';
import styles from './../../styles/header-footer.css';

class HeaderFooter extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    let movieOpend = this.props.movieOpend;
    let emptyResults = this.props.emptyResults;

    let headerFooterBody = null;
    if (movieOpend) {
      headerFooterBody = <MovieInfo />;
    } else if (!emptyResults) {
      headerFooterBody = <SearchNav />;
    }

    return (
      <div className = {styles.root}>
        <div className = {common.wrapper}>
          {headerFooterBody}
        </div>
      </div>
    );
  }

}

export default HeaderFooter;
