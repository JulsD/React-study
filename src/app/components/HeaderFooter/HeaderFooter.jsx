import React from 'react';

import Wrapper from './../Wrapper';
import styles from './header-footer.css';

class HeaderFooter extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className = {styles.root}>
        <Wrapper>
          {this.props.children}
        </Wrapper>
      </div>
    );
  }

}

export default HeaderFooter;
