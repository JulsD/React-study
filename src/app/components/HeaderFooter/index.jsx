import HeaderFooter from './HeaderFooter';
export default HeaderFooter;