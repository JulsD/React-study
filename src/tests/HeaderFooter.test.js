import React from 'react';
import HeaderFooter from '../app/components/HeaderFooter';

test('HeaderFooter should render a <div>', () => {
  const wrapper = shallow(
    <HeaderFooter />
  );
  expect(wrapper).toMatchSnapshot();
});
