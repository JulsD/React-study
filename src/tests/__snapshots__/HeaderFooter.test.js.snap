// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFooter should render a <div> 1`] = `
<div
  className="root"
>
  <Wrapper />
</div>
`;
